### DISCUSSION TITLE
Tests fail with recommended installation

### DISCUSSION THREAD

**Discussion by bilalcaliskan** [kubernetes-sigs/kubebuilder#2395] [Support] [answered]
Hello there, 

I have installed the Kubebuilder with the recommended approach mentioned in latest [Kubebuilder book](https://book.kubebuilder.io/quick-start.html#installation). It just installs the kubebuilder binary and moves it to **/usr/local/bin/**. 

So, when i run go test command in the root directory of my project, it gives me following output:
```shell
$ go test ./...
...
Failure [0.001 seconds]
[BeforeSuite] BeforeSuite 
/home/joshsagredo/go/src/github.com/bilalcaliskan/kubebuilder-tutorial/controllers/suite_test.go:66

  Unexpected error:
      <*fmt.wrapError | 0xc0006818a0>: {
          msg: "failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory",
          err: {
              Op: "fork/exec",
              Path: "/usr/local/kubebuilder/bin/etcd",
              Err: 0x2,
          },
      }
      failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory
  occurred
```

As you see, it searches the etcd binary in **/usr/local/kubebuilder/bin** directory but docs does not mention about it at all. With a little research, i have found that legacy kubebuilder book recommends [this installation method](https://book-v1.book.kubebuilder.io/getting_started/installation_and_setup.html) which installs kubebuilder as a folder under **/usr/local/kubebuilder** directory. I assume that with legacy installation, there is an etcd binary under **/usr/local/kubebuilder/bin**

Here is my **kubebuilder version** output:
```shell
$ kubebuilder version
Version: main.version{KubeBuilderVersion:"3.1.0", KubernetesVendor:"1.19.2", GitCommit:"92e0349ca7334a0a8e5e499da4fb077eb524e94a", BuildDate:"2021-05-27T17:54:28Z", GoOs:"linux", GoArch:"amd64"}
```

There are lots of conversations about that problem, i have found [that issue](https://github.com/chaos-mesh/chaos-mesh/issues/1282) which is closed but as i mentioned in that comment, with the latest Kubebuilder book, i have no **ensure-kubebuilder** make target in my Makefile.

How should i proceed? Should i try the legacy installation method or forget about testing with the assistance of latest docs :)

**Comment by camilamacedo86** [comment, marked as answer]
It happened because your local env has not had the envtest binaries.
the docs and scaffold with the makefile targets have this scenario sorted out in the latest releases so I am marking this one as answered.

**Comment by camilamacedo86** [comment]
Closing since it is answered.
